tests/geocoding_tests.py

```diff
@@ -15,62 +15,323 @@
         response = self.get_resp(url, json_=data)
         assert columns[0].get("name") in response
 
     def test_get_invalid_columns(self):
         url = "/geocoder/geocoding/columns"
-        tableDto = models.TableDto(10001, "no_table")
+        table_dto = models.TableDto(10001, "no_table")
+        data = {"table": table_dto.to_json()}
 
-        data = {"table": tableDto.to_json()}
         response = self.get_resp(url, json_=data)
 
-        message = "No columns found for table with name {0}".format(tableDto.name)
-        assert message in response
+        error_message = f"No columns found for table with name {table_dto.name}"
+        assert error_message in response
+
+    def test_does_valid_column_name_exist(self):
+        table_id = self.sqla_departments.id
+        columns = reflection.Inspector.from_engine(db.engine).get_columns(
+            self.sqla_departments.table_name
+        )
+        column_name = columns[0].get("name")
+
+        response = Geocoder()._does_column_name_exist(table_id, column_name)
+        assert True is response
+
+    def test_does_column_name_not_exist(self):
+        table_id = self.sqla_departments.id
+        column_name = "no_column"
+
+        response = Geocoder()._does_column_name_exist(table_id, column_name)
+        assert False is response
+
+    def test_does_table_not_exist(self):
+        table_id = -1
+        column_name = "no_column"
+
+        error_message = f"Table with ID {table_id} does not exists"
+        with self.assertRaisesRegex(TableNotFoundException, error_message):
+            Geocoder()._does_column_name_exist(table_id, column_name)
+
+    def test_load_data_from_all_columns(self):
+        table_id = self.sqla_departments.id
+        geo_columns = ["street", "city", "country"]
+
+        data = Geocoder()._load_data_from_columns(table_id, geo_columns)
+        assert 5 == len(data)
+        assert ("Oberseestrasse 10", "Rapperswil", "Switzerland") in data
+
+    def test_load_data_from_columns_with_none(self):
+        table_id = self.sqla_departments.id
+        geo_columns = ["street", None, "country"]
+
+        data = Geocoder()._load_data_from_columns(table_id, geo_columns)
+        assert 5 == len(data)
+        assert ("Oberseestrasse 10", "Switzerland") in data
+
+    def test_add_lat_lon_columns(self):
+        table = self.sqla_departments
+        lat_column_name = "latitude"
+        lon_column_name = "longitude"
+
+        columns = self.sqla_departments.columns
+        number_of_columns_before = len(columns)
+
+        Geocoder()._add_lat_lon_columns(table, lat_column_name, lon_column_name)
+
+        columns = self.sqla_departments.columns
+        number_of_columns_after = len(columns)
+
+        assert number_of_columns_after == number_of_columns_before + 2
+        column_names = [column.column_name for column in columns]
+        assert lon_column_name in column_names
+        assert lat_column_name in column_names
```

Comment:
several asserts in one test. (kinda makes sense here but could be a comment from reviewers)